src/System.IO.Pipelines/PipelineWriterExtensions.cs

```diff
@@ -9,7 +9,12 @@ public static class PipelineWriterExtensions
     {
         private readonly static Task _completedTask = Task.FromResult(0);
 
-        public static Task WriteAsync(this IPipeWriter output, Span<byte> source)
+        public static Task WriteAsync(this IPipeWriter output, byte[] source)
+        {
+            return WriteAsync(output, new ArraySegment<byte>(source));
+        }
+
+        public static Task WriteAsync(this IPipeWriter output, ArraySegment<byte> source)
```

Comment:
There isn't anything inherently wrong with having a WriteAsync overloads that take span. If you actually don't want this overload, then feel free to remove. The compiler won't complain if you had this method take a span since it isn't async. The tests were async.